### ISSUE TITLE
Make front end not explicitly rely on sessions

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#7747]
By default, we force implementations to use HTTPSessions.  We currently use them to store our language, locale and/or EDIT_MODE and time machine checks.

It is entirely do-able to remove this session requirement for visitors who are browsing in the default language,  This would greatly increase horizontal scalability.

**Comment by chrismccracken** [comment]
+1 I'd love to see this as a config option to enable/disable. Clustered sessions become significantly more feasible when scoped to backend requests only. This would need very clear docs on which frontend features/velocity depend on session state.

**Comment by brentgriffin** [comment]
I think this will work against the upcoming rules engine functionality

**Comment by wezell** [comment]
I don't disagree Brent, though it should affect only some of the rule Conditionlets.  You could still do things like segment via geolocation, browser, requested url, ip etc...

The no session thing is an important option for large sites looking to easily scale out horizontally.  I think we should should have a config variable that disables/enables sessions.   We have one kind of like that now called "ENABLE_CLICKSTREAM_TRACKING" which defaults to true though I didn't respect that in the code changes committed.  Maybe we should - if ENABLE_CLICKSTREAM_TRACKING=true, we create sessions.  If not, not?

**Comment by oarrietadotcms** [comment]
PR, now with ee dependency : https://github.com/dotCMS/core/pull/7748

**Comment by brentgriffin** [comment]
QA - should be able to test by verifying that there is no JSESSIONID cookie when clickstream functionality is turned off in properties file.  If you pass a language key on the URL or if you have clickstream enabled, you should have a JSESSIONID cookie.

**Comment by wezell** [comment]
right.  With ENABLE_CLICKSTREAM_TRACKING=true:

```
curl --head http://localhost:8080/
cabbage:dotcms3 will$ curl --head http://localhost:8080/
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Set-Cookie: dmid=bd838e45-926b-4cb3-a6e8-4ecc6cfcbfa1; Expires=Fri, 03-Apr-2020 16:16:00 GMT; Path=/
Set-Cookie: JSESSIONID=3BEA7796F7FE49ABF673A9BE150A4326; Path=/; HttpOnly
Content-Type: text/html;charset=UTF-8
Transfer-Encoding: chunked
Date: Wed, 20 May 2015 16:16:03 GMT
```
and with ENABLE_CLICKSTREAM_TRACKING=false::

```
cabbage:dotcms3 will$ curl --head http://localhost:8080/
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Set-Cookie: dmid=1a249615-7a0f-4231-bf00-769682548786; Expires=Fri, 03-Apr-2020 16:18:01 GMT; Path=/
Content-Type: text/html;charset=UTF-8
Transfer-Encoding: chunked
Date: Wed, 20 May 2015 16:18:04 GMT
```

**Comment by bryanboza** [comment]
Fixed, tested in the 3.2.1 release and works as expected...
